lando_cli/cli.py

```diff
@@ -229,17 +230,37 @@ def submit_to_lando(
     return
 
 
-def git_run(git_args: list[str], repo: Path) -> str:
+def git_run(*args, **kwargs) -> str:
     """Helper to run `git` with consistent arguments."""
+    return _git_run(*args, **kwargs)
+
+
+def git_run_bytes(*args, **kwargs) -> bytes:
+    """Helper to run `git` with consistent arguments, and return raw bytes output."""
+    kwargs["raw"] = True
+    return _git_run(*args, **kwargs)
+
+
+def _git_run(git_args: list[str], repo: Path, raw: bool = False):
+    """Helper to run `git` with consistent arguments.
+
+    If raw is True, data is returned as bytes. Otherwise, a string with normalised
+    newlines is returned.
+    """
     command = ["git", *git_args]
+    extra_run_args = {}
+    if not raw:
+        extra_run_args = {
+            "encoding": "utf-8",
+            "text": True,
+        }
     result = subprocess.run(
         command,
-        encoding="utf-8",
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
-        text=True,
         check=True,
         cwd=repo,
+        **extra_run_args,
     )
     return result.stdout.strip()
```

Comment:
Looks like we're running `.strip()` out the `_git_run` output. We should definitely not do this for `git_run_bytes`, I imagine that is the cause of the patch application failures we saw on dev. We may want to consider removing it from `git_run` as well, allowing the caller to decide if output needs to be stripped, or using some kwarg like `strip=True`.